src/lib/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,88 @@
+import styled from "@emotion/styled";
+import { useState } from "react";
+
+type BadgeSize = "small" | "large";
+
+export interface BadgeProps {
+  icon: string;
+  alt: string;
+  reverse: boolean;
+  label: string;
+  size: BadgeSize;
+}
+
+const badgeHeight = {
+  large: "24px",
+  small: "16px",
+};
+
+const StyledBadge = styled.div<Pick<BadgeProps, "reverse" | "size">>`
+  display: flex;
+  ${({ reverse }) => reverse && "flex-direction: row-reverse;"}
+  height: ${({ size }) => badgeHeight[size]};
+  border-radius: ${({ size }) => (size === "large" ? "4px" : "2px")};
+
+  background-color: ${({ theme }) => theme.colors.background.dark[200]};
+`;
+
+const BadgeIconWrapper = styled.div<Pick<BadgeProps, "size">>`
+  display: grid;
+  place-items: center;
+  width: ${({ size }) => badgeHeight[size]};
+  height: 100%;
+`;
+
+const iconSize = {
+  large: "12px",
+  small: "8px",
+};
+
+const BadgeIcon = styled.img<Pick<BadgeProps, "size">>`
+  width: ${({ size }) => iconSize[size]};
+  height: ${({ size }) => iconSize[size]};
+`;
+
+const StyledLabel = styled.div<Pick<BadgeProps, "size">>`
+  color: #f0f0f0;
```

Comment:
Please use colors from the theme, as it will allow us to implement configurable theming in the future without massive refactoring.